### DISCUSSION TITLE
io_uring_prep_recv_multishot - not clear what the parameters are for

### DISCUSSION THREAD

**Discussion by romange** [axboe/liburing#1018] [Q&A] [answered]
`io_uring_prep_recv_multishot` function has `buf` and `len` arguments but I was under the impression that the only way to use `io_uring_prep_recv_multishot` is together with with provided buffers. But the latter make these arguments obsolete. What's the reason for exposing these arguments?

**Comment by axboe** [comment, marked as answer]
For some multishot variants, setting len can be used to limit the receive side. But this actually isn't the case for recv multishot, it'll error if you do that. addr I agree never makes sense. Honestly not sure how they ended up being exposed, they should be NULL/0 always. Doesn't really matter further down as we set/clear them anyway regardless, it's just more of an API wart than anything.
